src/main/scala/chisel3/util/SizeOf.scala

```diff
@@ -0,0 +1,30 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package chisel3.util
```

Comment:
There is a PR open for intrinsics: https://github.com/chipsalliance/firrtl-spec/pull/43

Reply:
The intrinsic itself would still be `circt` specific though, right? Then it should live under the `circt` namespace.
We could also just add a `sizeOf` primitive op in firrtl instead of a `circit` specific intrinsic.

Reply:
Planning wise, this is one of many intriniscs.  The point of inttrinsics is to separate features which need compiler knowledge, and hence cannot be libraries, from language specification.   This PR is infrastructure for a capability, which will eventually be in the spec as an op.  It is for now done with legal firrtl.  Which is why I put this in chisel3: the output of using the intrinsic objects is valid firrtl now.  (just as SSE intriniscs are valid C functions, they aren't a new kind of thing, just a specific interpretation of an existing thing)

Reply:
I'm all for adding intrinsic emission support to chisel3. But i'm also confused about this specific sizeOf intrinsic. What's an expected use case for this particular one? Is this best as a test case that uses the chisel3-support-for-intrinsics?

Reply:
It is generally useful and a very standard meta-programming function when you have inferred widths or other cases of non-statically knowable types to correctly handle the result of inference.  Chisel has width inference, so being able to react (statically in the generated code, since you can't react to it in Chisel land) to the result of that inference is useful.  (for example, computing values to put on a bus based on the width in a type-safe way)

This is also a simple example of an intrinsic to get the ball rolling on the infrastructure.  It is a capability which requires compile support and cannot be done at chisel time or as a library (well, with width inference, I can think of a few ways, but they are super ugly).

Reply:
It makes sense in a near-future world where we can react to information after width inference Ints and Strings to potentially build hardware with widths that react to that value, but I don't think one could emit verilog that reacts to a sizeof value being driven on a wire, so I'm still not sure what the immediate use case is here.